### PR TITLE
[React Native] CustomCheckboxGroup Recipe

### PR DESCRIPTION
**Pull request by mcavo**
#### Add react native custom checkbox group component
* Features:
  * Single or multiple options
  * Set value selected by default
  * Set selected values
  * Set disabled values
  * Show error message

<p align="center">
  
</p>

#### Usage

``` ts
import CustomCheckboxGroup from '@components/CustomCheckboxGroup';

const optionsRecord = {
  A: 'option A',
  B: 'option B',
  C: 'option C',
  D: 'option D',
  E: 'option E',
};

function MyCheckboxGroup() {
  return (
    <CustomCheckboxGroup
      multioption
      title="Multiple option checkbox"
      options={optionsRecord}
      disabledOptions={['E']}
      selectedOptions={['A', 'B']}
    />
  );
}

export default MyCheckboxGroup;
```

#### Props
`<K extends React.Key = string, V = string>`

| Prop  | Default  | Type | Description |
| :------------ |:---------------:| :---------------:| :-----|
| checkboxContainerStyle | - | [ViewStyle](https://reactnative.dev/docs/view-style-props) | Checkbox container style, wrapping the check icon and the option label |
| checkboxStyle | - | [ViewStyle](https://reactnative.dev/docs/view-style-props) | Checkbox icon container style |
| checkboxTextProps | - | [TextProps](https://reactnative.dev/docs/text#props) | Checkbox option label text props |
| defaultOption | - | `K` | Selected option by default |
| disabledOptions | - | `Array<K>` | List of disabled options. |
| error | - | string | Error description. |
| multioption | `false` | `boolean` | Allows multiple selection of options. |
| onChange | - | `(selected: Array<K>) => void`| Function that recieves all the selected options keys and is called each time the selection list changes. |
| options | - | [Record<K, V>](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt) | Map with options keys as keys and option label as values. |
| selectedOptions | - | `Array<K>`| List of selected options, has priority over `defaultOption`.
| style | - | [ViewStyle](https://reactnative.dev/docs/view-style-props) | Checkbox group style |
| title | - | `string` | Checkbox group title. |

#### Pending
* We could add tests to all our components

PS: Let me know if you believe this needs another feature 🤗 

## Trello Card
https://trello.com/c/nsv7aebV/179-checkbox-group

### CODE REVIEW

File: cookbook-react-native/recipes/checkboxes/custom-checkboxes/index.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      [] as Array<K>
+    ));
+  }, [selected])
+
+  const handlePress = (option: K) =>
+    setSelected({
+      ...(multioption ? selected : {}),
+      [option]: !selected[option]
+    } as Record<K, boolean>);
+    console.tron.log(selected)
```
Comment (guidoprinc) [inline cookbook-react-native/recipes/checkboxes/custom-checkboxes/index.tsx:54]: this should be deleted?

Comment (mcavo) [reply cookbook-react-native/recipes/checkboxes/custom-checkboxes/index.tsx:54]: ups!

File: cookbook-react-native/recipes/checkboxes/custom-checkboxes/components/Checkbox/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const handlePress = () => onPress(optionKey);
+  const isSelected = !disabled && selected;
+  const backgroundColor = getBackgroundColor(isSelected);
+  const borderColor = getBorderColor(disabled);
+  return (
+    <TouchableOpacity style={[styles.container, style]} onPress={handlePress} disabled={disabled}>
+      <View style={[styles.iconContainer, { backgroundColor, borderColor }, checkboxStyle]}>
+        {isSelected && <Image source={checkIcon} style={styles.icon} />}
+      </View>
+      <CustomText gray={disabled} textProps={textProps} >{option}</CustomText>
```
Comment (sfernandez11) [inline cookbook-react-native/recipes/checkboxes/custom-checkboxes/components/Checkbox/index.tsx:39]: `<CustomText gray={disabled} textProps={textProps}>{option}</CustomText>`

File: cookbook-react-native/recipes/checkboxes/custom-checkboxes/constants.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  defaultValue?: T,
+  multioption?: boolean;
+  disabledOptions?: Array<T>;
+  onChange?: () => void;
+  error?: string;
+  checkboxContainerStyle?: ViewStyle;
+  checkboxStyle?: ViewStyle;
+  checkboxTextProps?: TextProps;
+  style?: ViewStyle;
+}
```
Comment (sfernandez11) [inline cookbook-react-native/recipes/checkboxes/custom-checkboxes/constants.ts:15]: This interface is repeated in the `index.tsx`, you can delete it

File: cookbook-react-native/recipes/checkboxes/custom-checkbox-group/readme.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# Cookbook React Native Bootstrap: CustomCheckboxGroup Recipe
```
Comment (sfernandez11) [inline cookbook-react-native/recipes/checkboxes/custom-checkbox-group/readme.md:1]: Amazing 👏 👏

**Pull request merged**